enumerables.rb

```diff
@@ -0,0 +1,143 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield (self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    selected = []
+    my_each { |i| selected << i if yield(i) }
+    selected
+  end
+
+  def my_all?
+    result = true
+    my_each { |i| result = false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each { |i| result = true if yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each { |i| result = false if yield(i) }
+    result
+  end
+
+  def my_count
+    result = 0
+    my_each { |i| result += 1 if yield(i) }
+    result
+  end
+
+  def my_map(arr = nil)
+    result = []
+    if arr.nil?
+      my_each { |i| result.push(yield(i)) }
+    else
+      my_each { |i| result.push result.call(i) }
+    end
+    result
+  end
+
+  def my_inject
+    result = first
```

Comment:
* [ ] There should be a working `my_inject`  identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
#### Let us compare the results return in each of the examples below and make the return the same result as the original method.
```p (5..10).inject(1) { |product, n| product * n }``` and ```p (5..10).my_inject(1) { |product, n| product * n }``` must return same result.